Agent/Services/FileLogsManager.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Remotely.Shared.Services;
+
+public interface IFileLogsManager
+{
+    Task<bool> AnyLogsExist();
+    Task DeleteLogs();
+    IAsyncEnumerable<byte[]> ReadAllBytes();
+}
+
+public class FileLogsManager : IFileLogsManager
+{
+    public async Task<bool> AnyLogsExist()
+    {
+        using var logLock = await FileLoggerDefaults.AcquireLock();
+
+        var componentName = Assembly.GetExecutingAssembly().GetName().Name;
+        var directory = Path.Combine(FileLoggerDefaults.LogsFolderPath, $"{componentName}");
+
+        if (Directory.Exists(directory))
+        {
+            foreach (var file in Directory.GetFiles(directory))
+            {
+                if (new FileInfo(file).Length > 0)
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public async Task DeleteLogs()
+    {
+        using var logLock = await FileLoggerDefaults.AcquireLock();
+
+        var componentName = Assembly.GetExecutingAssembly().GetName().Name;
+        var directory = Path.Combine(FileLoggerDefaults.LogsFolderPath, $"{componentName}");
+
+        if (Directory.Exists(directory))
+        {
+            foreach (var file in Directory.GetFiles(directory))
+            {
+                try
+                {
+                    File.Delete(file);
+                }
+                catch { }
+            }
+        }
+    }
+
+    public async IAsyncEnumerable<byte[]> ReadAllBytes()
```

Comment:
This now reads all files instead of only today's.  Returns `IAsyncEnumerable` so it doesn't have to load all files into memory at once.

Reply:
Good call.  I added it for all the methods in this class.